modules/core/shared/src/main/scala/gem/CoAdds.scala

```diff
@@ -0,0 +1,8 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem
+
+// CoAdds has a DISTINCT type due to its check constraint so we need a fine-grained mapping
+// here to satisfy the query checker.
+final case class CoAdds(toShort: Short)
```

Comment:
A couple of things here:

* The comment there applies to the DAO package version, not the model concept per se.  It requires this mapping because there is a database constraint that its value be `> 0`.

* We should guarantee that the model class is, correspondingly, always positive non-zero.  So I think the way we're doing this at least for now is with `sealed abstract case class` instances.  See `gem.math.Wavelength` which has the same deal.

* In the `CoAdds` companion we should define a `One` instance since that is the default, and an `Order` instance I expect along with a `fromShort: Prism[Short, CoAdds]` similar to the one for `Wavelength`.